pyang/plugins/sid.py

```diff
@@ -0,0 +1,566 @@
+            self.validate_sid()
+
+        self.set_module_information()
+        self.collect_module_items(module)
+        self.sort_items()
+        self.assign_sid()
+
+        if self.list_content:
+            self.list_all_items()
+        else:
+            self.list_deleted_items()
+
+        if self.check_consistency:
+            if self.is_consistent:
+                print("Check completed successfully")
+            else:
+                print("The .sid file need to be updated.")
+        else:
+            if self.is_consistent:
+                print("No .sid file generated, the current .sid file is already up to date.")
+            else:
+                self.generate_file()
+                if self.sid_file_created:
+                    print("\nFile %s created" % self.output_file_name)
+                else:
+                    print("\nFile %s updated" % self.output_file_name)
+                self.print_statistic()
+
+
+    ########################################################
+    def set_initial_range(self, range):
+        components = range.split(':')
+        if len(components) != 2 or not re.match(r'\d+:\d+', range):
+            raise SidParcingError("invalid range in argument, must be '<entry-point>:<size>'.")
+
+        self.content = OrderedDict([('assignment-ranges', [])])
+        self.content['assignment-ranges'].append(OrderedDict([('entry-point', int(components[0])), ('size', int(components[1]))]))
+
+    ########################################################
+    # Retrieve the module revision from the pyang context
+    def get_module_revision(self, module):
+        revision = None
+        for substmt in module.substmts:
+            if substmt.keyword == 'revision':
+                if revision == None:
+                    revision = substmt.arg
+                else:
+                    if revision < substmt.arg:
+                        revision = substmt.arg
+
+        if revision == None:
+            raise SidParcingError("no revision found in YANG definition file.")
+        return revision
+
+    ########################################################
+    # Set the 'module-name' and/or 'module-revision' in the .sid file if require
+    def set_module_information(self):
+        if 'module-name' not in self.content or self.module_name != self.content['module-name']:
+            self.content['module-name'] = self.module_name
+            if self.check_consistency == True:
+                print("ERROR, Mismatch between the module name defined in the .sid file and the .yang file.")
+                self.is_consistent = False
+
+        if 'module-revision' not in self.content or self.module_revision != self.content['module-revision']:
+            self.content['module-revision'] = self.module_revision
+            if self.check_consistency == True:
+                print("ERROR, Mismatch between the module revision defined in the .sid file and the .yang file.")
```

Comment:
Rather than printing errors directly, the idea is to register a plugin-specific error code, and set it.

Also, in general there is an error EmitError that should be used if possible to generate errors from the emit process (assumes you are using -f sid).